lib/cadet/assessments/question.ex

```diff
@@ -1,34 +1,37 @@
 defmodule Cadet.Assessments.Question do
   @moduledoc """
-  Questions model contains domain logic for questions management 
+  Questions model contains domain logic for questions management
   including programming and multiple choice questions.
   """
   use Cadet, :model
 
   alias Cadet.Assessments.Assessment
-  alias Cadet.Assessments.ProblemType
+  alias Cadet.Assessments.QuestionType
   alias Cadet.Assessments.QuestionTypes.ProgrammingQuestion
   alias Cadet.Assessments.QuestionTypes.MCQQuestion
+  alias Cadet.Assessments.QuestionTypes.Library
 
   schema "questions" do
     field(:title, :string)
     field(:display_order, :integer)
-    field(:weight, :integer)
     field(:question, :map)
-    field(:type, ProblemType)
+    field(:type, QuestionType)
     field(:raw_question, :string, virtual: true)
+    field(:max_xp, :integer)
+    embeds_one(:library, Library)
```

Comment:
We should add `:raw_library` for json deserialisation

Reply:
as discussed, no need.